server/src/server.ts

```diff
@@ -95,19 +97,22 @@ let findBinaryDirPathFromProjectRoot = (
     return null;
   }
 
-  return findBinaryDirPathFromProjectRoot(parentDir);
+  return findBinaryFromProjectDirRec(parentDir, targetDir, binaryName);
 };
 
-let getBinaryDirPath = (projectRootPath: p.DocumentUri) =>
-  extensionConfiguration.binaryPath == null
-    ? findBinaryDirPathFromProjectRoot(projectRootPath)
-    : extensionConfiguration.binaryPath;
-
 let findRescriptBinary = (projectRootPath: p.DocumentUri) =>
-  utils.findRescriptBinary(getBinaryDirPath(projectRootPath));
+  extensionConfiguration.binaryPath == null
+    ? findBinaryFromProjectDirRec(
+        projectRootPath,
+        c.nodeModulesBinDir,
+        c.rescriptBinName
+      )
+    : utils.findRescriptBinary(extensionConfiguration.binaryPath);
 
 let findBscBinary = (projectRootPath: p.DocumentUri) =>
```

Comment:
I think can just assume that the bsc binary is right next to the rescript binary.
As if it isn't, there's a big problem no matter what.
So this could be reflected in the logic: find the path to rescript, and adapt it to a path to bsc instead.
This would remove the additional call to `findBinaryFromProjectDirRec`

Reply:
With this PR, in a simple project made with `rescript init`, the paths would default to
`projectRoot/node_modules/.bsc/rescript` and
`projectRoot/node_modules/rescript/{platformDir}/bsc.exe` 

So they are not in the same folder. Except you mean `rescript.exe` of course.

Reply:
Let's avoid calling the 2 commands in different ways.
We have enough bugs already.

Reply:
I don't know what you mean? I should find the path to rescript.exe or rescript (the one that we use now)? I had to look for the actual `bsc.exe` and call it directly, because it won't work on Windows otherwise.

Or do you mean, if you have a rescript binary in `node_modules/.bsc` you probably have the `node_modules/rescript/{platformDir}/bsc.exe`?

Reply:
I was for a second hoping one could call `rescript.exe` but the `rescript` node script actually does useful work, so I guess we'll have to call that.
OK so let me revise:
- use `node_modules/.bin/rescript` as source of truth to determine where ReScript is installed
- once that is found, assume, as an implementation detail that can be changed in future that the executable is in `node_modules/rescript/{platformDir}/bsc.exe` so just use that directly by massaging the path to `rescript` already found.

So a single, documented resolution logic for `rescript`, plus a little accessory function for `bsc`.

Reply:
If we want to decouple the logic for finding `bsc.exe` in future, I think we can make `rescript` provide the path to `bsc.exe` as a specific command-line option, and use that once on project initialisation.

Reply:
> So a single, documented resolution logic for `rescript`, plus a little accessory function for `bsc`.
I think updated the PR to reflect what you mean.